tutorials/high_performance_networking/README.md

```diff
@@ -0,0 +1,2171 @@
+        2025-03-06 12:20:27 - WARNING - CPU 4: Governor is set to 'powersave', not 'performance'.
+        2025-03-06 12:20:27 - WARNING - CPU 5: Governor is set to 'powersave', not 'performance'.
+        2025-03-06 12:20:27 - WARNING - CPU 6: Governor is set to 'powersave', not 'performance'.
+        2025-03-06 12:20:27 - WARNING - CPU 7: Governor is set to 'powersave', not 'performance'.
+        2025-03-06 12:20:27 - WARNING - CPU 8: Governor is set to 'powersave', not 'performance'.
+        2025-03-06 12:20:27 - WARNING - CPU 9: Governor is set to 'powersave', not 'performance'.
+        2025-03-06 12:20:27 - WARNING - CPU 10: Governor is set to 'powersave', not 'performance'.
+        2025-03-06 12:20:27 - WARNING - CPU 11: Governor is set to 'powersave', not 'performance'.
+        ```
+
+=== "manual"
+
+    ```bash
+    cat /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor
+    ```
+
+    ??? abstract "See an example output"
+
+        In this example, all cores were defaulted to `powersave` instead of the recommended `performance`.
+
+        ```
+        powersave
+        powersave
+        powersave
+        powersave
+        powersave
+        powersave
+        powersave
+        powersave
+        powersave
+        powersave
+        powersave
+        powersave
+        ```
+
+Install `cpupower` to more conveniently set the governor:
+
+```bash
+sudo apt update
+sudo apt install -y linux-tools-$(uname -r)
+```
+
+Set the governor to `performance` for all cores:
+
+=== "One-time"
+
+    ```bash
+    sudo cpupower frequency-set -g performance
+    ```
+
+=== "Persistent"
+
+    ```bash
+    cat << EOF | sudo tee /etc/systemd/system/cpu-performance.service
+    [Unit]
+    Description=Set CPU governor to performance
+    After=multi-user.target
+
+    [Service]
+    Type=oneshot
+    ExecStart=/usr/bin/cpupower -c all frequency-set -g performance
+
+    [Install]
+    WantedBy=multi-user.target
+    EOF
+    sudo systemctl enable cpu-performance.service
+    sudo systemctl start cpu-performance.service
+    ```
+
+Running the checks above should now list `performance` as the governor for all cores. You can also run `sudo cpupower -c all frequency-info` for more details.
+
+### 3.7 Prevent the GPU from going idle
+
+Similarly to the above, we want to maximize the GPU's clock speed and prevent it from going idle.
+
+Enable persistence mode on your GPU:
+
+```bash
+sudo nvidia-smi -pm 1
+```
+
+??? abstract "See an example output"
+
+    ```
+    Enabled persistence mode for GPU 00000005:09:00.0.
+    All done.
+    ```
+
+You can confirm that the persistence mode is enabled with:
+
+```bash
+nvidia-smi -q | grep -i "Persistence Mode"
+```
+
+??? abstract "See an example output"
+
+    ```
+    Persistence Mode: Enabled
+    ```
+
+We then lock the GPU's clocks to the max values:
+
+
+=== "One-time"
+
+    ```bash
+    sudo nvidia-smi --lock-gpu-clocks=$(nvidia-smi --query-gpu=clocks.max.sm --format=csv,noheader,nounits)
```

Comment:
Could use the shorthand `-lmc` and `lgc` to make this a little smaller

Reply:
You know I've not settles on whether to use short or long form in the tutorial. technically you just copy paste it so long is nice and descriptive, but if we want people to remember it to run more often then short is nice (like `nvidia-smi -q` or `nvidia-smi topo -mp`).

Does that make you lean one way vs the other?